src/request.js

```diff
@@ -0,0 +1,237 @@
+let http = require('http');
+let https = require('https');
+let assert = require('assert');
+let urllib = require('url');
```

Comment:
Please consider importing this as `URL` or `url` and then use the variable name `u` for urls...

urllib reminds me of python horrors and scares me unnecessarily :)